integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/PodUtils.java

```diff
@@ -454,4 +458,55 @@ public static boolean checkPodLogContains(String matchStr, String podName, Strin
     return Kubernetes.getPodLog(podName,namespace,null).contains(matchStr);
 
   }
+
+  /**
+   * Check if the pod log contains the certain text.
+   * @param regex check string
+   * @param podName the name of the pod
+   * @param namespace the Operator namespace
+   * @return true if regex found, false otherwise.
+   */
+  public static boolean checkPodLogContainsRegex(String regex, String podName, String namespace) {
+    // get the Pod logs
+    String operatorPodLog = assertDoesNotThrow(() -> getPodLog(podName, namespace));
+
+    // match regex in pod log
+    getLogger().info("Search: {0} in pod {1} log", regex, podName);
+    Pattern pattern = Pattern.compile(regex);
+    Matcher matcher = pattern.matcher(operatorPodLog);
+
+    return matcher.find();
+  }
+
+  /**
+   * Search specified regex in an uncompleted pod log.
+   * @param regex check string
+   * @param domainUid domain uid of the domain
+   * @param namespace the Operator namespace
+   * @return true if regex found, false otherwise.
+   */
+  public static boolean checkInUncompletedIntroPodLogContainsRegex(String regex, String domainUid, String namespace) {
+    // get introspector pod name
+    String introspectorPodName = assertDoesNotThrow(()
+        -> getIntrospectorPodName(domainUid, namespace), "Getting introspector pod name failed");
+    getLogger().info("introspector pod name is: {0}", introspectorPodName);
+
+    // get the domain status message
+    StringBuffer getPodLogCmd = new StringBuffer("kubectl logs pod/");
```

Comment:
You can use TestActions.getPodLog() directly here.

Reply:
I first tried TestActions.getPodLog() but it didn't work for this case, this method gets logs from a completed pod but this case needs logs from an incomplete pod :(